File: platform/College-1/Litcoder/Week 2/Code/lab5.java
import java.util.Scanner;

public class Main {
    private boolean isOdd(int num) {
        return (num % 2) != 0;
    }

    private int findSingleSum(int num) {
        // Base Case
        if (num < 10) {
            return num;
        }
        int sum = 0;
        int rem;
        while (num != 0) {
            rem = num % 10;
            sum += rem;
            num = num / 10;
        }
        num = sum;
        return findSingleSum(num);
    }

    private char returnChar(int num) {
        char[] oddToChar = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i'};
        return oddToChar[num - 1];
    }

    public static void main(String[] args) {
        Main ob = new Main();
        Scanner input = new Scanner(System.in);
        String[] inputs = input.nextLine().split(" ");
        String outStr = "";
        for (int index = 0; index < inputs.length; index++) {
            int currentNumber = Integer.parseInt(inputs[index]);
            currentNumber = ob.findSingleSum(currentNumber);
            if (ob.isOdd(currentNumber)) {
                outStr += ob.returnChar(currentNumber);
            } else {
                outStr += currentNumber;
            }
        }
        System.out.print(outStr);
        input.close();
    }
}

File: platform/College-1/Litcoder/Week 2/Code/lab5.py
def isOdd(num):
    return num % 2 != 0

def findSum(num):
    # Base Case
    if num < 10:
        return num
    sum = 0
    while num != 0:
        rem = num % 10
        sum = sum + rem
        num = int(num / 10)
    return findSum(sum)

def findChar(num):
    char = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i']
    return char[num - 1]

str_input = input().split(' ')
outStr = ""

for num in str_input:
    currentNum = int(num)
    currentNum = findSum(currentNum)
    if isOdd(currentNum):
        outStr = f"{outStr}{findChar(currentNum)}"
    else:
        outStr = f"{outStr}{currentNum}"

print(outStr, end='')
